### PR TITLE
Include updated model architecture diagram in developer guide and update use cases

### PR DESCRIPTION
**Pull request by Caleblyx**
Update model architecture diagram in developer guide to include the new task package and other new components.

### CODE REVIEW

File: docs/diagrams/UpdatedBetterModelClassDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+UniqueTagList -[hidden]down- UniquePersonList
+UniqueTagList -[hidden]down- UniquePersonList
+
+
+UniqueTagList *-right-> "*" Tag
+UniquePersonList o-right-> Person
+
+Person -up-> "*" Tag
+
+Person *--> Name
```
Comment (luciatirta) [inline docs/diagrams/UpdatedBetterModelClassDiagram.puml]: perhaps you want to consider adding multiplicity for the attributes of a person object to make it consistent with the rest of the diagram

File: docs/diagrams/UpdatedModelClassDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+Model .right.> ObservableList
+ModelManager --> "1" AddressBook
+ModelManager --> "1" TaskList
+ModelManager -left-> "1" UserPrefs
+UserPrefs .up.|> ReadOnlyUserPrefs
+
+
+AddressBook *--> "1" UniquePersonList
+UniquePersonList --> "*" Person
+Person *--> Name
```
Comment (luciatirta) [inline docs/diagrams/UpdatedModelClassDiagram.puml]: same comment as above

**Review by luciatirta — COMMENTED** [review]
Great diagram! perhaps you want to make the diagram more consistent by including the multiplicity for person object

**Review by lerxcl — APPROVED** [review]
Looks good to merge after the changes made! Good job 👍

**Pull request merged**